lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -298,6 +297,31 @@ public void onAuthUpdated() {
 		}
 	}
 
+	/**
+	 * Called when where was an error during authentication attempt
+	 *
+	 * @param errorInfo Error associated with unsuccessful authentication
+	 */
+	public void onAuthError(ErrorInfo errorInfo) {
+		Log.i(TAG, String.format("onAuthError: (%d) %s", errorInfo.code, errorInfo.message));
+		switch (state.state) {
+			case connecting:
+				ITransport transport = this.transport;
+				if (transport != null)
+					/* onTransportUnavailable will send state change event and set transport to null */
+					onTransportUnavailable(transport, null, errorInfo);
```

Comment:
Now I think of it, why do this at all? The server will disconnect the transport with the relevant error if it thinks that its auth params are not valid. I don't think there is any need for this here. It is possible that the transport is already using a new, valid, token, for example, and some operation elsewhere just failed and triggered the `onAuthError()`